innovation.game.php

```diff
@@ -3232,6 +3254,79 @@ function checkForSpecialAchievementsForPlayer($player_id) {
             throw $e; // Re-throw the flag
         }
     }
+
+    /** Checks to see if any players gain or lose any flag/fountain achievements. **/
+    function updateFlagsAndFountains() {
+        $end_of_game = false;
+
+        foreach (self::getActivePlayerIdsInTurnOrderStartingWithCurrentPlayer() as $player_id) {
+            $opponent_ids = self::getActiveOpponentIds($player_id);
+            for ($color = 0; $color < 5 ; $color++) {
+                // Flags
+                $num_visible_flags = self::countVisibleIconsInPile($player_id, 8 /* flag */, $color);
+                $num_visible_cards = self::countVisibleCards($player_id, $color);
+                $oppoent_has_more_visible_cards = false;
+                foreach ($opponent_ids as $opponent_id) {
+                    if (self::countVisibleCards($opponent_id, $color) > $num_visible_cards) {
+                        $oppoent_has_more_visible_cards = true;
+                    }
+                }
+                $desired_flag_achievements = $oppoent_has_more_visible_cards ? 0 : $num_visible_flags;
```

Comment:
opponent isn't spelled right here.

Reply:
Good catch! I'll fix this in a follow up.